service/src/main.rs

```diff
@@ -738,45 +776,82 @@ fn enclave_account<E: EnclaveBase>(enclave_api: &E) -> AccountId32 {
 }
 
 // Alice plays the faucet and sends some funds to the account if balance is low
-fn ensure_account_has_funds(api: &mut Api<sr25519::Pair, WsRpcClient>, accountid: &AccountId32) {
+fn ensure_account_has_funds(
+	api: &mut Api<sr25519::Pair, WsRpcClient>,
+	accountid: &AccountId32,
+) -> Result<(), Error> {
+	// check account balance
+	let free = api.get_free_balance(accountid)?;
+	info!("TEE's free balance = {:?}", free);
+
+	let existential_deposit = api.get_existential_deposit()?;
+	info!("Existential deposit is = {:?}", existential_deposit);
+	let funding_amount = existential_deposit
```

Comment:
When reading this function, it takes some time to understand the calculation and why this is done here.
To make it easier for someone else to understand the code, you could do the following:
- add explanatory variable names. Like  `maximum_funding_amount = existential_deposit.saturating_mul(MIN_FUND_INCREASE_FACTOR)`
 and `amount_to_be_funded = maximum_funding_amount.saturating_sub(current_enclave_balance)`
- add an extra function with an explanatory name . Same principle as above, but instead of a variable allocation, you can just call `let amount_to_be_funded = diff_between_maximum_funding_amount_and_current_free_balance()` or a mix of all , like l `let amount_to_be_funded = maximum_funding_amount.saturating_sub(get_current_enclave_balance())` 

That just as an example - if you have a better proposal - go for it!